tasks/LinearTemperatureSensorTask.cpp

```diff
@@ -0,0 +1,80 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "LinearTemperatureSensorTask.hpp"
+
+using namespace raw_io;
+using namespace base::samples;
+
+LinearTemperatureSensorTask::LinearTemperatureSensorTask(std::string const& name)
+    : LinearTemperatureSensorTaskBase(name)
+{
+}
+
+LinearTemperatureSensorTask::~LinearTemperatureSensorTask()
+{
+}
+
+// Convert the voltage in temperature
+Temperature temperature(base::Time const& timestamp,
+    double const& voltage,
+    TemperatureSensorSettings const& sensor_config)
+{
+    return Temperature::fromCelsius(timestamp,
```

Comment:
And if you did use base::Temperature instead of 'float' for temperatures, there wouldn't be the need for this fromCelsius (which, given it was not documented in the settings class, is very counter-intuitive given that we normalize all quantities to SI units)